cache.go

```diff
@@ -56,10 +52,7 @@ func (p *publicKeysCache) loadAffineIntoCache(key []byte, affine *blst.P1Affine)
 	if !enabledCache {
 		return
 	}
-	p.mu.Lock()
-	defer p.mu.Unlock()
-	copy(p.publicKeySharedBuffer[:], key)
-	p.publicKeyCache[p.publicKeySharedBuffer] = *affine
+	p.publicKeyCache.Store([48]byte(key), *affine)
```

Comment:
[48]byte(key) causes an allocation, can you use `p.publicKeySharedBuffer`?

Reply:
Are you sure about that? I didn't know 🤔 

How can I verfy that it really allocates memory?